test/e2e/framework/util.go

```diff
@@ -1048,7 +1014,8 @@ type KubectlBuilder struct {
 // NewKubectlCommand returns a KubectlBuilder for running kubectl.
 func NewKubectlCommand(args ...string) *KubectlBuilder {
 	b := new(KubectlBuilder)
-	b.cmd = KubectlCmd(args...)
+	tk := e2ekubectl.NewTestKubeconfig(TestContext.CertDir, TestContext.Host, TestContext.KubeConfig, TestContext.KubeContext, TestContext.KubectlPath)
+	b.cmd = tk.KubectlCmd(args...)
 	return b
 }
```

Comment:
The dependency on e2ekubctl here is fine.
The methods that depend on `NewKubctlCommand()` will be moved at a later point as part of #84615
The methods that depend on this one in this file are:
* RunKubectlOrDie 
* RunKubectl 
* RunKubectlOrDieInput 
* RunKubectlInput